covalent_dispatcher/_core/execution.py

```diff
@@ -296,133 +337,128 @@ def task_callback(future: Future):
     result_object._status = Result.RUNNING
     result_object._start_time = datetime.now(timezone.utc)
 
-    order = result_object.lattice.transport_graph.get_topologically_sorted_graph()
+    g = result_object.lattice.transport_graph._graph
+
+    app_log.debug(f"Beginning workflow {result_object.dispatch_id}")
+
+    for node_id, d in g.in_degree():
+        app_log.debug(f"Node {node_id} has {d} parents")
+
+        pending_deps[node_id] = d
+        tasks_left += 1
+        if d == 0:
+            tasks_queue.put(node_id)
+
+    while tasks_left > 0:
+        app_log.debug(f"{tasks_left} tasks left")
+
+        tasks_left -= 1
+        node_id = tasks_queue.get()
+        app_log.debug(f"Processing node {node_id}")
+
+        if node_id < 0:
+
+            app_log.debug(f"Workflow {result_object.dispatch_id} failed or cancelled.")
+            wait(futures)
+            return result_object
 
-    for nodes in order:
-        futures: list = []
+        node_name = result_object.lattice.transport_graph.get_node_value(node_id, "name")
 
-        for node_id in nodes:
-            # Get name of the node for the current task
-            node_name = result_object.lattice.transport_graph.get_node_value(node_id, "name")
+        if node_name.startswith(
+            (subscript_prefix, generator_prefix, parameter_prefix, attr_prefix)
+        ):
+            if node_name.startswith(parameter_prefix):
+                output = result_object.lattice.transport_graph.get_node_value(node_id, "value")
+            else:
+                parent = result_object.lattice.transport_graph.get_dependencies(node_id)[0]
+                output = result_object.lattice.transport_graph.get_node_value(parent, "output")
 
-            if node_name.startswith(
-                (subscript_prefix, generator_prefix, parameter_prefix, attr_prefix)
-            ):
-                if node_name.startswith(parameter_prefix):
-                    output = result_object.lattice.transport_graph.get_node_value(node_id, "value")
+                if node_name.startswith(attr_prefix):
```

Comment:
@cjao can you confirm that you intend for this to be a 3x nested conditional nested inside a while loop?

Reply:
@scottwn Yes.